pkgs/applications/editors/sublime/4/common.nix

```diff
@@ -23,7 +23,7 @@ let
     "x86_64-linux" = "x64";
   }.${stdenv.hostPlatform.system};
 
-  libPath = lib.makeLibraryPath [ xorg.libX11 xorg.libXtst glib libglvnd openssl gtk3 cairo pango ];
+  libPath = lib.makeLibraryPath [ xorg.libX11 xorg.libXtst glib libglvnd openssl gtk3 cairo pango curl.out ];
```

Comment:
I wonder why you used `curl.out` instead of just `curl`

Reply:
Hi @turboMaCk, I already had this discussion with SuperSandro2000 here: https://github.com/NixOS/nixpkgs/pull/146934#pullrequestreview-819459216 I already tried all of the curl package's outputs and only curl.out has the library files in them. If there is some way of doing this that wasn't already discussed in that thread please let me know.

Reply:
I think you must have made some mistake when trying that. I tried it before commenting and retried it now again and it seems to work as expected. if you use just `curl` it works. Either that or perhaps https://github.com/NixOS/nixpkgs/pull/147804 fixed this in some mysterious way.

This is the patch I made to your PR:

```diff
diff --git a/pkgs/applications/editors/sublime/4/common.nix b/pkgs/applications/editors/sublime/4/common.nix
index 777ee84a39b..81eaa442425 100644
--- a/pkgs/applications/editors/sublime/4/common.nix
+++ b/pkgs/applications/editors/sublime/4/common.nix
@@ -23,7 +23,7 @@ let
     "x86_64-linux" = "x64";
   }.${stdenv.hostPlatform.system};
 
-  libPath = lib.makeLibraryPath [ xorg.libX11 xorg.libXtst glib libglvnd openssl gtk3 cairo pango curl.out ];
+  libPath = lib.makeLibraryPath [ xorg.libX11 xorg.libXtst glib libglvnd openssl gtk3 cairo pango curl ];
 in let
   binaryPackage = stdenv.mkDerivation {
     pname = "${pname}-bin";
```

and test
```
$ NIXPKGS_ALLOW_UNFREE=1 nix-build -A sublime4
/nix/store/0jnq3rdl62d69f0vc0d2zq90g1rs7jk6-sublimetext4-4121
$ ./result/bin/subl  
```

note that there are no more errors.

Reply:
`lib.makeLibraryPath` will try `lib` output if not explicitly specified, and fall back on `out` if that fails:

https://github.com/NixOS/nixpkgs/blob/7fd3e067cd2929e1c0b59287b5b799ad7450f59d/lib/attrsets.nix#L490-L491

Reply:
right thanks @jtojnar I expected something like this would be the case. Anyway this is one more reason to use just `curl` otherwise build of this will break once we fix curl to put shared object to lib rather than out.

Reply:
also related https://github.com/NixOS/nixpkgs/pull/150564/

Reply:
@jtojnar Thanks for clarifying that, I had no idea. I new that the default "curl" output did not have the libs and so would not work but I didn't realize makeLibraryPath had some magic inside that made it use curl.out instead. The pull request has been updated.